apps/api/scripts/backfill-internal-tx-claims.ts

```diff
@@ -0,0 +1,446 @@
+      try {
+        // Decode the full event data
+        const decodedLog = decodeEventLog({
+          abi: [claimedEventAbi],
+          data: log.data as `0x${string}`,
+          topics: log.topics as [`0x${string}`, ...`0x${string}`[]],
+        }) as unknown as {
+          args: {
+            account: string;
+            amount: bigint;
+            claimedAmount: bigint;
+            season: number;
+          };
+        };
+
+        const claim = {
+          account,
+          blockNumber: log.blockNumber,
+          transactionHash: log.transactionHash,
+          amount: decodedLog.args.amount,
+          claimedAmount: decodedLog.args.claimedAmount,
+          season: decodedLog.args.season,
+        };
+
+        missingClaims.push(claim);
+        totalEligibleAmount += BigInt(claim.amount);
+        totalClaimedAmount += BigInt(claim.claimedAmount);
+      } catch {
+        console.log(`⚠️  Failed to decode event at block ${log.blockNumber}`);
+      }
+    }
+  }
+
+  // Display results
+  if (missingClaims.length === 0) {
+    console.log(
+      `${colors.green}✅ No missing claims found in block range ${startBlock} to ${endBlock}!${colors.reset}`,
+    );
+    console.log(
+      `\n💡 All Claimed events in this range are already indexed in the database.`,
+    );
+    return;
+  }
+
+  console.log(
+    `${colors.yellow}⚠️  Found ${missingClaims.length} missing claim events!${colors.reset}\n`,
+  );
+
+  // Show the full list of missing claims
+  for (const claim of missingClaims) {
+    console.log(`${colors.red}Missing Claim:${colors.reset}`);
+    console.log(`  Account: ${claim.account}`);
+    console.log(`  Block: ${claim.blockNumber}`);
+    console.log(`  Tx Hash: ${claim.transactionHash}`);
+    console.log(`  Amount:         ${claim.amount}`);
+    console.log(`  Claimed Amount: ${claim.claimedAmount}`);
+    console.log(`  Season: ${claim.season}`);
+    console.log();
+  }
+
+  // Analyze the transactions to understand why they're missing
+  console.log(
+    `${colors.blue}📈 Analyzing missing transactions...${colors.reset}\n`,
+  );
+
+  const missingTxHashes = new Set(missingClaims.map((c) => c.transactionHash));
+
+  console.log("Missing Claim txn hashes:");
+  for (const hash of missingTxHashes) {
+    console.log(" - ", hash);
+  }
+  console.log("");
+
+  console.log(`\n${colors.magenta}Summary:${colors.reset}`);
+  console.log(`  Total missing claims: ${missingClaims.length}`);
+  console.log(
+    `  Total eligible amount: ${(Number(totalEligibleAmount) / 1e18).toFixed(2)} tokens`,
+  );
+  console.log(
+    `  Total claimed amount: ${(Number(totalClaimedAmount) / 1e18).toFixed(2)} tokens`,
+  );
+  console.log(
+    `  Average claim percentage: ${totalEligibleAmount > 0n ? ((Number(totalClaimedAmount) * 100) / Number(totalEligibleAmount)).toFixed(2) : 0}%`,
+  );
+  console.log("");
+
+  // If --update flag is provided, insert missing claims into database
+  if (shouldUpdate) {
```

Comment:
As shown in the package.json scripts, if the `--update` flag is set the database will be synced.  If not the logs show you how many Claim events are found that don't have a record in the db.